File: src/frontend/src/app/pages/secure/order/interfaces/orders-get-single-response.ts
export interface OrdersGetSingleDeliveryResponse {
    id: string;
    name: string;
    creationDate: string;
    isDelivered: boolean;
    tripStarted: boolean;
    isLost: boolean;
    delivererFullName: string;
    delivererEmail: string;
    delivererPhoneNumber: string;
    products: OrdersGetSingleProductResponse[];
}

export interface OrdersGetSingleProductResponse {
    name: string;
    quantity: number;
    unit: string;
}

export interface OrdersGetSingleResponse {
    id: string;
    name: string;
    percentage: number;
    creationDate: string;
    countryName: string;
    gpsLatitude: number;
    gpsLongitude: number;
    city: string;
    postalCode: string;
    fullStreet: string;
    deliveries: OrdersGetSingleDeliveryResponse[];
    products: OrdersGetSingleProductResponse[];
}

File: src/backend/Orders/Actions/OrdersActions/OrdersGetSingle/OrdersGetSingleResponse.cs
namespace Orders.Actions.OrdersActions.OrdersGetSingle
{
    public class OrdersGetSingleResponse
    {
        public string Id { get; set; } = default!;
        public string Name { get; set; } = default!;
        public int Percentage { get; set; } = default!;
        public DateTime CreationDate { get; set; } = default!;
        public string CountryName { get; set; } = default!;
        public double GpsLatitude { get; set; }
        public double GpsLongitude { get; set; }
        public string City { get; set; } = default!;
        public string PostalCode { get; set; } = default!;
        public string FullStreet { get; set; } = default!;
        public List<OrdersGetSingleDeliveryResponse> Deliveries { get; set; } = default!;
        public List<OrdersGetSingleProductResponse> Products { get; set; } = default!;
    }
}
